echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/PipelineCache.java

```diff
@@ -101,33 +130,37 @@ public int getPollingIntervalSeconds() {
   }
 
   /**
-   * Returns an observable that emits the configured pipelines as of the most recent polling cycle.
-   * If no polling cycles have been completed, the observable will wait until the first cycle
-   * completes and then emit the pipelines from that polling cycle.
-   *
-   * @return An observable emitting the pipelines as of the most recent polling cycle
+   * The list of pipelines as of the last successful polling cycle, or null if not available yet
    */
-  public Observable<List<Pipeline>> getPipelines() {
-    return pipelineSubject.take(1);
+  @Nullable
+  public List<Pipeline> getPipelines() {
+    return pipelines;
   }
 
-  public List<Pipeline> getPipelinesSync() {
-    return getPipelines().toBlocking().first();
+  @Nonnull
+  public List<Pipeline> getPipelinesSync() throws TimeoutException {
+    return getPipelinesSync(TimeUnit.SECONDS.toMillis(pollingIntervalSeconds));
```

Comment:
I'm not sure that `pollingIntervalSeconds` is the correct maximum time to wait here.  In general, this is resolving the case where Echo starts processing events before the `PipelineCache` starts polling.  The time we'd have to wait is then:
```(time between event processor starting and pipeline cache starting) + (time to get a response back to a single poll)``` which is independent of the time between polls.

That being said, reasonable values of the polling interval (~10-60 seconds) are probably also reasonable values for this timeout, so the end result is probably fine.

Reply:
Yes I just tried to use a semi-reasonable default for callers who don't care too much. The motivation for using `pollingIntervalSeconds` is that it guarantees at least one polling cycle has occurred while we're waiting